src/AzureAuth/CommandMain.cs

```diff
@@ -347,9 +347,35 @@ public int OnExecute()
             // Small bug in Lasso - Add does not accept a null IEnumerable here.
             this.eventData.Add("settings_scopes", this.authSettings.Scopes ?? new List<string>());
 
+            if (this.InteractivityDisabled())
+            {
+                this.eventData.Add("interactivity_disabled", true);
+                this.logger.LogCritical("Interactive authentication is disabled.");
+
+                return 1;
+            }
+
             return this.ClearCache ? this.ClearLocalCache() : this.GetToken();
         }
 
+        /// <summary>
+        /// Determines whether the given env is user interactive or not.
+        /// </summary>
+        /// <returns>A boolean to indicate user interactive env</returns>
+        public bool InteractivityDisabled()
+        {
+            var disableInteractivity = this.env.Get(EnvVars.DisableInteractivity);
+            var corextNonInteractive = this.env.Get(EnvVars.CorextNonInteractive);
+
+            if (!string.IsNullOrEmpty(disableInteractivity) ||
+                string.Equals("1", corextNonInteractive))
+            {
+                return true;
+            }
+
+            return false;

```

Comment:
Generally when you have an `if` conditional that returns `true` or `false` you can simplify to return the boolean expression instead of using an `if` with multiple `returns`.